src/dotnet/APIView/APIViewWeb/Managers/CopilotCommentsManager.cs

```diff
@@ -31,163 +31,136 @@ IConfiguration configuration
+                document.BadCode = badCode;
+                document.Embedding = embedding;
             }
 
-            if (comment != null)
+            if (goodCode != null)
             {
-                update = update.Set("Comment", comment);
+                document.GoodCode = goodCode;
             }
 
-            if (guidelineIds != null)
+            if (comment != null)
             {
-                update = update.Set("GuidelineIds", guidelineIds);
+                document.Comment = comment;
             }
 
-            if (badCode != null)
+            if (guidelineIds != null)
             {
-                var embedding = await GetEmbeddingsAsync(badCode);
-                update = update
-                    .Set("BadCode", badCode)
-                    .Set("Embedding", embedding);
+                document.GuidelineIds = SplitGuidelineIds(guidelineIds);
             }
 
-            return await _copilotCommentsRepository.UpdateDocumentAsync(filter, update);
+            await _copilotCommentsRepository.UpdateDocumentAsync(document);
+            return GetDocumentJsonWithoutEmbedding(document);
         }
 
-        public Task DeleteDocumentAsync(string user, string id)
+        public Task DeleteDocumentAsync(string id, string language, string user)
         {
-            var filter = GetIdFilter(id);
-
-            var updateBuilder = Builders<CopilotCommentModel>.Update;
-            var update = updateBuilder
-                .Set("IsDeleted", true)
-                .Set("ModifiedOn", DateTime.UtcNow)
-                .Set("ModifiedBy", user);
+            return _copilotCommentsRepository.DeleteDocumentAsync(id, language, user); 
+        }
 
-            return _copilotCommentsRepository.DeleteDocumentAsync(filter, update); 
+        public async Task<string> GetDocumentAsync(string id, string language)
+        {
+            var document = await _copilotCommentsRepository.GetDocumentAsync(id, language);
+            return GetDocumentJsonWithoutEmbedding(document);
         }
 
-        public async Task<string> GetDocumentAsync(string id)
+        public async Task<string> SearchDocumentsAsync(string language, string badCode, float threshold, int limit)
         {
-            var filter = GetIdFilter(id);
-            var document = await _copilotCommentsRepository.GetDocumentAsync(filter);
+            var embeddings = await GetEmbeddingsAsync(badCode);
+            var documents = await _copilotCommentsRepository.SearchLanguage(language);
 
-            var documentJson = JsonConvert.SerializeObject(document);
+            var searchResults = new List<CopilotSearchModel>();
+            foreach (var document in documents)
+            {
+                var similarity = CosineSimilarity(document.Embedding, embeddings);
+                if (similarity < threshold)
+                    continue;
+                searchResults.Add(new CopilotSearchModel(similarity, document));
+            }
```

Comment:
Agree, can't we just search the database directly? It looks like we are just getting everything for a given language and then manually comparing every document, which seems like it would negate the benefit of having a DB search index.

Reply:
From what I understand so far, the way to add a search index in Cosmos is using Azure Cognitive Search. I can optimize it a bit using language as a partition, which we can do natively in Cosmos DB. Though it might be that I haven't fully grasped the concept of a DB search index and all the functionalities in Cosmos. Most of this code is [referenced from a previous cosmos database search](https://github.com/Azure/azure-sdk-tools/blob/086f9e22a83d9ed1f832ea9aee18605330ea030d/src/dotnet/APIView/APIViewWeb/Repositories/CosmosCommentsRepository.cs#L25) where they search for the `reviewId` and `lineId`. I can continue to look into it.